pkg/util/graph/graph.go

```diff
@@ -38,3 +40,37 @@ func GetServicePrincipalIDByAppID(ctx context.Context, graph *msgraph.GraphServi
 		return nil, fmt.Errorf("%d service principals have appId '%s'", len(matches), appId)
 	}
 }
+
+// GetServicePrincipalExpiryByAppID returns a service principal's expiry date from
+// an application (client) ID.
+func GetServicePrincipalExpiryByAppID(ctx context.Context, graph *msgraph.GraphServiceClient, appId string) (*string, error) {
```

Comment:
nit: return a `time.Time` here and then let whatever is calling this function manipulate and format it to what it needs.

Reply:
Done